src/modules/commander/state_machine_helper.cpp

```diff
@@ -368,9 +368,13 @@ main_state_transition(const vehicle_status_s &status, const main_state_t new_mai
 
 	case commander_state_s::MAIN_STATE_AUTO_MISSION:
 
+		if (!status_flags.condition_auto_mission_available) {
```

Comment:
What does this change? From above you can see `ret` stays `TRANSITION_DENIED` if it's not changed which means it was already denied if you had no valid mission.